py/desispec/spectra.py

```diff
@@ -336,7 +345,10 @@ def select(self, nights=None, bands=None, targets=None, fibers=None, invert=Fals
         ret = Spectra(keep_bands, keep_wave, keep_flux, keep_ivar, 
             mask=keep_mask, resolution_data=keep_res, 
             fibermap=self.fibermap[keep], meta=self.meta, extra=keep_extra,
-            single=self._single)
+            single=self._single, scores=keep_scores)
```

Comment:
@armengau, for consideration: you commented that `Spectra.__getitem__` was already handling filtering on `extra_catalog`, which made me realize that we have a maintenance headache here of having to keep track of which member variables to filter in both `__getitem__` and `select`.  I suggest that `select` should be updated to just construct `keep_rows` and then call `return self[keep_rows]` to do the actual subselection so that we only have one place to take care of the special cases (scores, extra, extra_catalog, ...).

Reply:
@sbailey: I agree, but `select` can also select over bands. So using `__getitem__` cannot work straightforward.

- Option 1: change `__getitem__(self, index, bands=None)`. Then `Spectra[...]` still works in principle and `select` would call `self.__getitem__(keep, bands=None)`

- Option 2 if this is not a good thing to do (I'm not so familiar with conventions/rules on `__xxx__` functions in classes): define a function `get_slice [or another name](self, index, bands=None)`, then redefine `__getitem__(self,index): return self.get_slice(index)`, and use `get_slice` in `select`

Reply:
@armengau either option 1 or option 2 is fine with me, go ahead and do whichever seems more natural/easy for you.

Reply:
Done, using option 2 for safety, as it seems clearly recommanded that __getitem__ always gets a single argument.